### PR TITLE
Og discovery

### PR DESCRIPTION
**Pull request by llzx373**
add Discovery support for openGuass

    1. if openGauss Primary changed, switch primary in Shardingsphere.
    2. DAL could add Datasource to openGauss Discovery Datasorce, and active
       for read loadbalance.
    3. if one of standbys failed,it will be disabled, read loadbalance continue
     without failed.

### CODE REVIEW

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/pom.xml [hunk 1]
```diff
@@ -31,5 +31,6 @@
     
     <modules>
         <module>shardingsphere-db-discovery-mgr</module>
+        <module>shardingsphere-db-discovery-og</module>
```
Comment (dongzl) [inline shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/pom.xml:34]: I think `og` is not a good name. ^_^

Comment (llzx373) [reply shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/pom.xml:34]: shardingsphere-db-discovery-openGauss？ any suggest?

Comment (dongzl) [reply shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/pom.xml:34]: Now, the exist module' name is `opengauss`, for example `shardingsphere-sql-parser-opengauss`,  `shardingsphere-db-protocol-opengauss`.

I suggest we can keep the same name.

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-og/src/main/java/org/apache/shardingsphere/dbdiscovery/og/OpenGaussDatabaseDiscoveryType.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        String newPrimaryDataSource = determinePrimaryDataSource(activeDataSourceMap);
+        if (newPrimaryDataSource.isEmpty()) {
+            return;
+        }
+        if (!newPrimaryDataSource.equals(oldPrimaryDataSource)) {
+            oldPrimaryDataSource = newPrimaryDataSource;
+            ShardingSphereEventBus.getInstance()
+                    .post(new PrimaryDataSourceChangedEvent(schemaName, groupName, newPrimaryDataSource));
+        }
+
```
Comment (menghaoranss) [inline shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-og/src/main/java/org/apache/shardingsphere/dbdiscovery/og/OpenGaussDatabaseDiscoveryType.java:100]: unnecessary empty line

File: shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-og/src/main/java/org/apache/shardingsphere/dbdiscovery/og/OpenGaussDatabaseDiscoveryType.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        JobConfigurationPOJO result = new JobConfigurationPOJO();
+        result.setJobName(jobName);
+        result.setCron(cron);
+        result.setShardingTotalCount(1);
+        return result;
+    }
+
+    @Override
+    public String getType() {
+        return "OpenGauss";
```
Comment (menghaoranss) [inline shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-og/src/main/java/org/apache/shardingsphere/dbdiscovery/og/OpenGaussDatabaseDiscoveryType.java:193]: Other module named `openGauss` , I suggest keep the same name.

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**